okta/resource_okta_threat_insight_settings.go

```diff
@@ -27,6 +30,42 @@ func resourceThreatInsightSettings() *schema.Resource {
 				Optional:    true,
 				Description: "List of Network Zone IDs to exclude to be not logged or blocked by Okta ThreatInsight and proceed to Sign On rules evaluation",
 				Elem:        &schema.Schema{Type: schema.TypeString},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					// k will be "network_excludes.#"", "network_excludes.1", ..., "network_excludes.N"
+					items := strings.Split(k, ".")
+					// special case before the resource has been created
+					if old == "" || (len(items) > 1 && items[1] == "#") {
```

Comment:
I think we can get rid of the `items := strings.Split(k, ".")` line and keep the conditional small
`if old == "" || "network_excludes.#" {`